py3status/modules/systemd_suspend_inhibitor.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""
+Turn on and off systemd suspend inhibitor.
+
+Configuration parameters:
+    format: display format for this module
+        (default '[\?color=state SUSPEND [\?if=state OFF|ON]]')
+    lock_types: specify state to inhibit, comma separated list
+        https://www.freedesktop.org/wiki/Software/systemd/inhibit/
+        (default 'handle-lid-switch,idle,sleep')
+    thresholds: specify color thresholds to use
+        (default [(True, 'bad'), (False, 'good')])
+
+Format placeholders:
+    {state} systemd suspend inhibitor state, eg True, False
+
+Color thresholds:
+    xxx: print a color based on the value of `xxx` placeholder
+```
+
+
+Examples:
+# display SUSPEND ON/OFF
+systemd_suspend_inhibitor {
+    format = '[\?color=state SUSPEND [\?if=state OFF|ON]]'
+    thresholds = [(True, "bad"), (False, "good")]
+}
+```
+
+@author Cyrinux https://github.com/cyrinux
+@license BSD
+
+SAMPLE OUTPUT
+[{'full_text': 'SUSPEND ON', 'color': '#00FF00'}]
+
+off
+[{'full_text': 'SUSPEND OFF', 'color': '#FF0000'}]
+"""
+
+from dbus import SystemBus
+from os import close
+
+
+STRING_DBUS_EXCEPTION = "DBUS error, systemd-logind not started?"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    format = "[\?color=state SUSPEND [\?if=state OFF|ON]]"
+    lock_types = "handle-lid-switch,idle,sleep"
+    thresholds = [(True, "bad"), (False, "good")]
+
+    def post_config_hook(self):
+        bus = SystemBus()
+        self.lock = None
+        self.lock_types = self.lock_types.replace(" ", "").replace(",", ":")
```

Comment:
(nit) `SystemBus().get_object(`l... to kill a line? And try first before initializing `self.lock*` stuff.

I'm not sure about...
>lock_types: specify state to inhibit, comma separated list

... as traditionally, we would want a list... and join them later with commas, colons, or whatever else for commands, urls, etc... In the past without `post_config_hook`, it was often a string of something separate... like this.

Reply:
True, i fix this